plugins/org.eclipse.n4js/src/org/eclipse/n4js/external/NpmPackageToProjectAdapter.java

```diff
@@ -311,26 +314,11 @@ IStatus addTypeDefinitions(File packageRoot, PackageJson packageJson, String mai
 		}
 
 		File packageVersionedN4JSDProjectRoot = new File(packageN4JSDsRoot, closestMatchingVersion.toString());
-		File packageVersionedN4JSDSrcFolder = new File(packageVersionedN4JSDProjectRoot, N4JSD_SRC_FOLDER);
 		try {
-			/*
-			 * Changed the computation of the source and target path for the N4JSD files as follows for IDE-2429.
-			 *
-			 * The .n4jsd files are assumed to be in a folder called "src" within the N4JSD project root folder.
-			 *
-			 * Their target path is computed from the package root path and the main module path. The given main module
-			 * path contains the file name (without extension) of the main module of the NPM, potentially prepended by a
-			 * path that represents folders. An example would be "lib/index". The target path for the N4JSD files is
-			 * then assumed to be in a folder called "lib" within the package root path.
-			 */
-			Path sourcePath = packageVersionedN4JSDSrcFolder.toPath();
+			Path sourcePath = packageVersionedN4JSDProjectRoot.toPath();
```

Comment:
how about different approach discussed in IDE-2429:
 - the N4JSD repo contains API projects with definitions
 - installed npm packages could have just `manifest.mf` generated with IMPL project type.
 - API-IMPL mechanisms in IDE take care of shadowing / runtime

this way we could avoid polluting npm with definition files (still need manifest though.)
Also this makes it easier to maintain n4jsds, i.e. runtime always works against IMPL, while IDE links against API. We avoid hammering npm package with n4jsd files.

Reply:
Yes, I agree in principle that it's nicer. As discussed, I do remember that it didn't work when I tested it for IDE-2429, but I don't remember the reason. I don't think we should introduce this change as part of the PR in any case.